.github/workflows/build.yaml

```diff
@@ -0,0 +1,224 @@
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: '${{ matrix.python }}'
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-${{ matrix.puppet }}-{{ matrix.python }}-pip-unit-${{ hashFiles('**/requirements*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.puppet }}-${{ matrix.ruby }}-pip-unit-
+      - name: Python prep
+        run: |
+          pip install virtualenv
+      - name: Test
+        run: 'make ${{ matrix.make_target }}'
+
+  integration:
+    name: '${{matrix.name}} - puppet (${{matrix.puppet}})'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # note: actions/setup-ruby only allows using a major.minor release of ruby
+          - ruby: '2.5'
+            name: 'centos7-puppet5'
+          - ruby: '2.5'
+            name: 'centos7-puppet6'
+          - ruby: '2.5'
+            name: 'ubuntu16-puppet5'
+          - ruby: '2.5'
+            name: 'ubuntu16-puppet6'
+          - ruby: '2.5'
+            name: 'ubuntu18-puppet5'
+          - ruby: '2.5'
+            name: 'ubuntu18-puppet6'
+    env:
+      CHECK: '${{ matrix.check }}'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '${{ matrix.ruby }}'
+      - name: Bundle prep
+        run: |
+          gem install bundler
+          bundle -v
+          rm -f Gemfile.lock
+          # Update system gems if requested. This is useful to temporarily workaround troubles in the test runner"
+          # See https://github.com/puppetlabs/pdk-templates/commit/705154d5c437796b821691b707156e1b056d244f for an example of how this was used"
+          # Ignore exit code of SIGPIPE'd yes to not fail with shell's pipefail set"
+          [ -z "$RUBYGEMS_VERSION" ] || (yes || true) | gem update --system $RUBYGEMS_VERSION
+          gem --version
+          bundle -v
+          bundle config path vendor/bundle
+          bundle config without 'system_tests'
+          # note: using custom gemfile for kitchen tests
+          bundle config gemfile build/kitchen/Gemfile
+          bundle lock
+        # restore cache AFTER doing 'bundle lock' so that Gemfile.lock exists
+      - uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-${{ matrix.name }}-${{ matrix.ruby }}-gems-integration-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.name }}-${{ matrix.ruby }}-gems-integration-
+      - name: Bundle install
+        run: |
+          bundle install --jobs $(nproc) --retry 3 
+      - name: Test
+        run: |
+          bundle exec kitchen test --debug ${{ matrix.name }}
+
+  deploy:
+    name: 'deploy to forge'
+    needs:
+      - unit
+      - integration
```

Comment:
:+1: 

Does it make sense to do something similar for the Integration tests?
Eg. start them only when lint checks and unit tests succeeded?

That can potentially save some computing resources if there was an issue with lint/unit.
Just trying to be a good OSS citizen :)

Reply:
That's fair, fixed!

I also noticed that i was not including the `unit-python` in this dependency graph.!